### PR TITLE
app-vim/jedi: version bump 0.10.0, bug #718262

### PR DESCRIPTION
**Pull request by TheChymera**
Package-Manager: Portage-2.3.99, Repoman-2.3.22
Signed-off-by: Horea Christian <chr@chymera.eu>

### CODE REVIEW

File: app-vim/jedi/jedi-0.10.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python{3_6,3_7,3_8} )
+
+inherit vim-plugin python-r1
```
Comment (juippis) [inline app-vim/jedi/jedi-0.10.0.ebuild:8]: Just by looking at the ebuild and installed files... I don't think *python-r1.eclass* is the correct one here. I'd use *python-any-r1.eclass* instead.

File: app-vim/jedi/jedi-0.10.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+KEYWORDS="~amd64 ~x86"
+REQUIRED_USE=${PYTHON_REQUIRED_USE}
+IUSE="test"
+
+RDEPEND="
+	${PYTHON_DEPS}
+	dev-python/jedi[${PYTHON_USEDEP}]
+	"
+BDEPEND="
+	app-editors/vim[python]
```
Comment (juippis) [inline app-vim/jedi/jedi-0.10.0.ebuild:24]: This is also RDEPEND.

File: app-vim/jedi/jedi-0.10.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	test? ( dev-python/pytest )
+	"
+
+S="${WORKDIR}/jedi-vim-${PV}"
+
+# Tests are broken as they require online connectivity.
+RESTRICT="test"
```
Comment (juippis) [inline app-vim/jedi/jedi-0.10.0.ebuild:31]: Well they were broken even with `FEATURES="-network-sandbox"`.

**Comment by TheChymera** [comment]
@juippis haven't used this eclass before, looks good to you?

**Comment by TheChymera** [comment]
Any idea what's up with the CI issue? Tried to fix this with the following ebuild, but I get an error:

```console
chymera@silenthost ~/src/gentoo/app-vim/jedi $ cat jedi-0.10.0.ebuild
# Copyright 1999-2020 Gentoo Authors
# Distributed under the terms of the GNU General Public License v2

EAPI=7

PYTHON_COMPAT=( python{3_6,3_7,3_8} )

inherit vim-plugin python-any-r1

DESCRIPTION="vim plugin: binding to the autocompletion library jedi"
HOMEPAGE="https://github.com/davidhalter/jedi-vim"
SRC_URI="https://github.com/davidhalter/jedi-vim/archive/${PV}.tar.gz -> ${P}.tar.gz"

LICENSE="MIT"
KEYWORDS="~amd64 ~x86"
IUSE="test"

DEPEND="${PYTHON_DEPS}"
RDEPEND="
	app-editors/vim[python]
	dev-python/jedi[${PYTHON_USEDEP}]
	"
BEDPEND="test? ( dev-python/pytest )"

S="${WORKDIR}/jedi-vim-${PV}"

# Tests are broken.
RESTRICT="test"

# Makefile tries hard to call tests so let's silence this phase.
src_compile() { :; }

src_install() {
	vim-plugin_src_install
}

src_test() {
	pytest -vv || die
}
chymera@silenthost ~/src/gentoo/app-vim/jedi $ ebuild jedi-0.10.0.ebuild manifest
Appending /home/chymera/src/gentoo to PORTDIR_OVERLAY...
Error(s) in metadata for 'app-vim/jedi-0.10.0':
  RDEPEND: Invalid atom (Invalid use dep: ''), token 2
```

File: app-vim/jedi/jedi-0.10.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+HOMEPAGE="https://github.com/davidhalter/jedi-vim"
+SRC_URI="https://github.com/davidhalter/jedi-vim/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+RDEPEND="
+	${PYTHON_DEPS}
+	dev-python/jedi[${PYTHON_USEDEP}]
```
Comment (juippis) [inline app-vim/jedi/jedi-0.10.0.ebuild:20]: Using *python-any-r1* eliminates the need for `PYTHON_USEDEP`. Instead you want to use 
`$(python_gen_cond_dep 'dev-python/jedi[${PYTHON_MULTI_USEDEP}]')`

File: app-vim/jedi/jedi-0.10.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	"
+RDEPEND="app-editors/vim[python]"
+BEDPEND="test? ( dev-python/pytest )"
```
Comment (ekohl) [inline app-vim/jedi/jedi-0.10.0.ebuild:23]:
```suggestion
BDEPEND="test? ( dev-python/pytest )"
```

File: app-vim/jedi/jedi-0.10.0.ebuild (hunk 4 above)
Comment (TheChymera) [reply app-vim/jedi/jedi-0.10.0.ebuild:20]: If I use that I get:

```
Appending /home/chymera/src/gentoo to PORTDIR_OVERLAY...
 * ERROR: app-vim/jedi-0.10.0::gentoo failed (depend phase):
 *   External commands disallowed while sourcing ebuild: python_gen_cond_dep dev-python/jedi[${PYTHON_MULTI_USEDEP}]
 *
 * Call stack:
 *            ebuild.sh, line 609:  Called source '/home/chymera/src/gentoo/app-vim/jedi/jedi-0.10.0.ebuild'
 *   jedi-0.10.0.ebuild, line  21:  Called command_not_found_handle 'python_gen_cond_dep' 'dev-python/jedi[${PYTHON_MULTI_USEDEP}]'
 *            ebuild.sh, line  89:  Called die
 * The specific snippet of code:
 *   		die "External commands disallowed while sourcing ebuild: ${*}"
 *
 * If you need support, post the output of `emerge --info '=app-vim/jedi-0.10.0::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=app-vim/jedi-0.10.0::gentoo'`.
 * Working directory: '/usr/lib64/python3.6/site-packages'
 * S: '/var/tmp/portage/app-vim/jedi-0.10.0/work/jedi-0.10.0'
```

Comment (juippis) [reply app-vim/jedi/jedi-0.10.0.ebuild:20]: Right, it's `python_gen_any_dep` with this eclass.

File: app-vim/jedi/jedi-0.10.0.ebuild (hunk 5 above)
Comment (TheChymera) [reply app-vim/jedi/jedi-0.10.0.ebuild]: @ekohl thanks for spotting this!

**Comment by TheChymera** [comment]
@juippis now? :)

File: app-vim/jedi/jedi-0.10.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+LICENSE="MIT"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+RDEPEND="
+	${PYTHON_DEPS}
+	dev-python/jedi[${PYTHON_USEDEP}]
+	"
+RDEPEND="app-editors/vim[python]"
```
[the remark this answers is not in the document]
Comment (ekohl) [reply app-vim/jedi/jedi-0.10.0.ebuild:22]: I think this is still relevant.

**Pull request closed without merging**